lib/identity_cache/configuration_dsl.rb

```diff
@@ -291,13 +291,9 @@ def ensure_cacheable_association(association, options)
         if association_reflection.options[:through]
           raise UnsupportedAssociationError, "caching through associations isn't supported"
         end
-        child_class = association_reflection.klass
-        scope = association_reflection.scope
-        if scope && !child_class.all.instance_exec(&scope).joins_values.empty?
-          raise UnsupportedAssociationError, "caching association scoped with a join isn't supported"
-        end
         options[:inverse_name] ||= association_reflection.inverse_of.name if association_reflection.inverse_of
         options[:inverse_name] ||= self.name.underscore.to_sym
+        child_class = association_reflection.klass
```

Comment:
It was moved back, so basically a revert of https://github.com/Shopify/identity_cache/pull/259/files#diff-6f09f93a419109ba1acdc9d10e897513 for this file